back-end/src/main/java/index/Indexer.java

```diff
@@ -37,52 +38,72 @@
+                        boolean isCitus = (Config.database == Config.Database.CITUS);
+                        if (Config.indexingScheme == Config.IndexingScheme.POSTGIS_SPATIAL_INDEX)
+                            indexer = PsqlSpatialIndexer.getInstance(isCitus);
+                        else if (Config.indexingScheme == Config.IndexingScheme.TILE_INDEX)
+                            indexer = PsqlTileIndexer.getInstance(isCitus);
+                        else if (Config.indexingScheme
+                                == Config.IndexingScheme.PSQL_NATIVEBOX_INDEX)
+                            indexer = PsqlNativeBoxIndexer.getInstance(isCitus);
+                        else if (Config.indexingScheme
+                                == Config.IndexingScheme.PSQL_NATIVECUBE_INDEX)
+                            indexer = PsqlCubeSpatialIndexer.getInstance();
+                        else
+                            throw new Exception(
+                                    "Index type "
+                                            + Config.indexingScheme.toString()
+                                            + " not supported for PSQL.");
+                    } else if (Config.database == Config.Database.MYSQL) {
+                        if (l.getIndexerType().equals("AutoDDInMemoryIndexer"))
+                            throw new Exception("AutoDD is not supported by MySQL indexers.");
+                        else if (l.getIndexerType().equals("PsqlPredicatedTableIndexer"))
+                            throw new Exception(
+                                    "PredicatedTable is not supported by MySQL indexers.");
+                        else if (Config.indexingScheme == Config.IndexingScheme.MYSQL_SPATIAL_INDEX)
+                            indexer = MysqlSpatialIndexer.getInstance();
+                        else if (Config.indexingScheme == Config.indexingScheme.TILE_INDEX)
+                            indexer = MysqlTileIndexer.getInstance();
+                        else
+                            throw new Exception(
+                                    "Index type "
+                                            + Config.indexingScheme.toString()
+                                            + "not supported for MySQL.");
+                    }
                 }
 
                 // associate indexer
-                Layer l = c.getLayers().get(layerId);
                 l.setIndexer(indexer);
+                l.setIndexerType(indexer.getClass().getSimpleName());
 
                 // pre-run getColumnNames, see issue #84: github.com/tracyhenry/kyrix/issues/84
                 l.getTransform().getColumnNames();
             }
     }
 
+    public static Indexer getIndexerByType(String type) {
+        try {
+            Class c = Class.forName("index." + type);
+            Method m = c.getMethod("getInstance");
+            return (Indexer) m.invoke(null);
+        } catch (ClassNotFoundException e) {
+            e.printStackTrace();
```

Comment:
Let's not throw an exception when the type is an empty string (since that's a valid value). Otherwise currently the code throws horrifying exceptions when I run the NBA/USMap example.

Reply:
I think this is due to the `e.printStackTrace()` ? Removing it will reduce the display of the exception. It will still work fine.